src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -575,6 +596,10 @@ private void startCleaner(long recordOffset, SnowflakeTelemetryPipeCreation pipe
                     e.getMessage(),
                     e.getStackTrace());
                 telemetryService.reportKafkaConnectFatalError(e.getMessage());
+                if (maxCleanerRetries >= 0 &&  pipeStatus.cleanerRestartCount.intValue() == maxCleanerRetries) {
+                  failedCleanerException.set(e);
+                  break;
+                }
                 forceCleanerFileReset = true;
```

Comment:
This will be unreachable if above `if` case is true and we will never call `resetCleanerFiles` right?

Reply:
Yup. If we've retried the max amount of times, we shouldn't do a forced cleaner file reset (which happens for each retry).